clients/src/test/java/org/apache/kafka/clients/consumer/internals/ConsumerCoordinatorTest.java

```diff
@@ -1413,10 +1458,55 @@ public void testOnJoinPrepareWithOffsetCommit() throws InterruptedException {
         time.sleep(150);
         res = coordinator.onJoinPrepare(pollTimer, generationId, memberId);
         assertFalse(res);
+        pollTimer = time.timer(100L);
+        time.sleep(150);
+        client.respond(offsetCommitResponse(singletonMap(t1p, Errors.UNKNOWN_MEMBER_ID)));
+        res = coordinator.onJoinPrepare(pollTimer, generationId, memberId);
+        assertTrue(res);
+
+        assertFalse(client.hasPendingResponses());
+        assertFalse(client.hasInFlightRequests());
+        assertFalse(coordinator.coordinatorUnknown());
+    }
+
+    @Test
+    public void testOnJoinPrepareWithOffsetCommit_KeepJoinAfterRebalanceTimeout() {
```

Comment:
nit: In Kafka, we usually don't have underscore `_` in the test name. Could we replace it with `should`, ex: `testOnJoinPrepareWithOffsetCommitShouldKeepJoinAfterRebalanceTimeout()`, and other tests also apply. Thanks.